perses/app/setup_relative_calculation.py

```diff
@@ -634,26 +665,44 @@ def run_setup(setup_options, serialize_systems=True, build_samplers=True):
                     _logger.info('SETUP FOR FAH DONE')
                     return {'topology_proposals': top_prop, 'hybrid_topology_factories': htf}
 
+                # get platform
+                platform = get_openmm_platform(platform_name=None)
+                # Setup context caches for multistate samplers
+                energy_context_cache = cache.ContextCache(capacity=None, time_to_live=None, platform=platform)
+                propagation_context_cache = cache.ContextCache(capacity=None, time_to_live=None, platform=platform)
+
                 if setup_options['fe_type'] == 'sams':
-                    hss[phase] = HybridSAMSSampler(mcmc_moves=mcmc.LangevinSplittingDynamicsMove(timestep=timestep,
-                                                                                        collision_rate=1.0 / unit.picosecond,
-                                                                                        n_steps=n_steps_per_move_application,
-                                                                                        reassign_velocities=False,
-                                                                                        n_restart_attempts=20,constraint_tolerance=1e-06,
-                                                                                        context_cache=cache.ContextCache(capacity=None, time_to_live=None)),
-                                                   hybrid_factory=htf[phase], online_analysis_interval=setup_options['offline-freq'],
-                                                   online_analysis_minimum_iterations=10,flatness_criteria=setup_options['flatness-criteria'],
-                                                   gamma0=setup_options['gamma0'])
-                    hss[phase].setup(n_states=n_states, n_replicas=n_replicas, temperature=temperature,storage_file=reporter,lambda_protocol=lambda_protocol,endstates=endstates)
+                    hss[phase] = HybridSAMSSampler(mcmc_moves=mcmc.LangevinSplittingDynamicsMove(
+                        timestep=timestep,
+                        collision_rate=1.0 / unit.picosecond,
+                        n_steps=n_steps_per_move_application,
+                        reassign_velocities=False,
+                        n_restart_attempts=20, constraint_tolerance=1e-06,
+                        context_cache=cache.ContextCache(capacity=None, time_to_live=None)),
```

Comment:
We no longer provide a `context_cache` argument to `MCMCMove`. Recall that we now [issue a warning if this happens](https://github.com/choderalab/openmmtools/blob/main/openmmtools/mcmc.py#L149-L152) since we now feed the `ContextCache` argument to `MCMCMove.apply()`.